File: example/SingularityObjCSample/SingularityObjCSample/SingularityViewController.m
//
//  ViewController.m
//  SingularityObjCSample
//
//  Created by Shubham Garg on 28/12/22.
//

#import "SingularityViewController.h"

@interface SingularityViewController ()
@end

@implementation SingularityViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.
}


- (IBAction)loginBtnAxn:(id)sender {
    UIViewController *vc;
    for (UIWindowScene *scene in [UIApplication sharedApplication].connectedScenes) {
        if (scene.activationState == UISceneActivationStateForegroundActive) {
            for (UIWindow *window in scene.windows) {
                if (window.isKeyWindow) {
                    vc = window.rootViewController;
                    break;
                }
            }
            break;
        }
    }
    NSString *jsonString = @"{\"singularityConfig\":{\"environment\":0},\"key\":\"neobrix\"}";
    NSData *data = [jsonString dataUsingEncoding:NSUTF8StringEncoding];
    NSDictionary* params = [NSJSONSerialization JSONObjectWithData:data options:0 error:nil];
    [SingularitySDKInitializer startLoginOn:vc with:params listener:self];
}

- (void)onGetSingularityUserInfoWithUser:(NSDictionary<NSString *,id> * _Nonnull)user {
    NSLog(@"user :: %@", user);
}

- (void)onSingularityClose {
    printf("close");
}

- (void)onSingularityErrorWithMessage:(NSString * _Nonnull)message code:(NSInteger)code {
    NSLog(@"message :: %@", message);
}

- (void)onSingularityLogout {
    printf("logout");
}

@end

File: example/SingularitySwiftSample/SingularitySwiftSample/SingularityViewController.swift
//
//  SingularityViewController.swift
//  SingularitySwiftSample
//
//  Created by Shubham Garg on 23/11/22.
//

import UIKit
import SingularityKit

class SingularityViewController: UIViewController {
    @IBOutlet weak var keyTF: UITextField!
    @IBOutlet weak var envTF: UITextField!
    @IBOutlet weak var loginBtn: UIButton!
    
    override func viewDidLoad() {
        super.viewDidLoad()
        setUPDefaultParams()
    }
    
    func setUPDefaultParams() {
        keyTF.text = "neobrix"
        envTF.text = "0"
    }

    @IBAction func loginBtnAxn(_ sender: Any) {
        let key = keyTF.text ?? "neobrix"
        var env = SingularityEnvironMent.test
        if envTF.text == "1" {
            env = .production
        }
        let params = SingularityParams(merchantId: key, environment: env)
        SingularitySDKInitializer.startLogin(on: self, with: params, listener: SingularityEventListner())
    }
}

class SingularityEventListner: SingularityListener {
    func onGetSingularityUserInfo(user: [String : Any]) {
        print(user)
    }
    
    func onSingularityClose() {
        print("onSingularityClose")
    }
    
    func onSingularityLogout() {
        print("onSingularityLogout")
    }
    
    func onSingularityError(message: String, code: Int) {
        print(message)
    }
    
    
}
